src/main/java/org/myrobotlab/service/InMoov2.java

```diff
@@ -1092,83 +1098,113 @@ public void onStarted(String name) {
-      if (actualName.equals(name)) {
-        AbstractSpeechRecognizer ear = (AbstractSpeechRecognizer) Runtime.getService(actualName);
-        ear.attachTextListener(getPeerName("chatBot"));
-      }
-
-      actualName = getPeerName("mouth");
-      if (actualName.equals(name)) {
-        AbstractSpeechSynthesis mouth = (AbstractSpeechSynthesis) Runtime.getService(actualName);
-        mouth.attachSpeechListener(getPeerName("ear"));
-      }
-
-      actualName = getPeerName("chatBot");
-      if (actualName.equals(name)) {
-        ProgramAB chatBot = (ProgramAB) Runtime.getService(actualName);
-        chatBot.attachTextListener(getPeerName("htmlFilter"));
-        startPeer("htmlFilter");
-      }
-
-      actualName = getPeerName("htmlFilter");
-      if (actualName.equals(name)) {
-        TextPublisher htmlFilter = (TextPublisher) Runtime.getService(actualName);
-        htmlFilter.attachTextListener(getPeerName("mouth"));
-      }
-
-      // FIXME - this is a much better way to do it than the code above
-      // the types are not exposed, its not using a local reference of the
-      // service
-      // its remote compatible, and it uses the peers actual name - it does not
-      // break easily,
-      // allows user modification and subscribing to topics for extensibility,
-      // no NPEs uses messaging, and follows mrl pub/sub conventions - above
-      // code should be
-      // refactored accordingly.
-      actualName = getPeerName("head");
-      if (actualName.equals(name)) {
-        addListener("publishMoveHead", actualName);
-      }
-
-      actualName = getPeerName("torso");
-      if (actualName.equals(name)) {
-        addListener("publishMoveTorso", actualName);
-      }
-
-      // mapping a channel to a single end point depending on peer
-      actualName = getPeerName("leftHand");
-      if (actualName.equals(name)) {
-        addListener("publishMoveLeftHand", actualName, "onMoveHand");
-      }
-
-      // mapping a channel to a single end point depending on peer
-      actualName = getPeerName("rightHand");
-      if (actualName.equals(name)) {
-        addListener("publishMoveRightHand", actualName, "onMoveHand");
+      if (peerKey == null) {
+        // service not a peer
+        return;
       }
 
-      // mapping a channel to a single end point depending on peer
-      actualName = getPeerName("leftArm");
-      if (actualName.equals(name)) {
-        addListener("publishMoveLeftArm", actualName, "onMoveArm");
+      if (runtime.isProcessingConfig() && !configStarted) {
+        invoke("publishEvent", "CONFIG STARTED " + runtime.getConfigName());
+        configStarted = true;
       }
 
-      // mapping a channel to a single end point depending on peer
-      actualName = getPeerName("rightArm");
-      if (actualName.equals(name)) {
-        addListener("publishMoveRightArm", actualName, "onMoveArm");
+      invoke("publishEvent", "STARTED " + peerKey);
+
+      switch (peerKey) {
```

Comment:
From what I can tell, this is autowiring peers together. Can this not be handled by the config system?

Reply:
yup .. exactly ..
this was started before config was capable .. it should be removed, but that will come in a later pr